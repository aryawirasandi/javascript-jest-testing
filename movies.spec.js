import movies from "./movies.js";

describe('Favorite Movies', () => {
    let myMovies = [];
    beforeEach(() => {
         myMovies = [
            {
                title : "Age of ultron",
                rate : null,
            },
        ]    
    });

    test('can add movie', () => {
        movies.add(myMovies, "Kungfu Panda")
        expect(myMovies).toMatchSnapshot();
    })

    test('rate a movie', () => {
        movies.rate(myMovies[0], 5);
        movies.add(myMovies, "Kungfu Panda")
        movies.rate(myMovies[1], 2);
        expect(myMovies).toMatchSnapshot();
    })
})