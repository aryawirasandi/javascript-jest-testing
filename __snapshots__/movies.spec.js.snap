// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite Movies can add movie 1`] = `
Array [
  Object {
    "rate": null,
    "title": "Age of ultron",
  },
  Object {
    "rate": null,
    "title": "Kungfu Panda",
  },
]
`;

exports[`Favorite Movies rate a movie 1`] = `
Array [
  Object {
    "rate": 5,
    "title": "Age of ultron",
  },
  Object {
    "rate": 2,
    "title": "Kungfu Panda",
  },
]
`;
